orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/PackageInfo.java

```diff
@@ -69,132 +87,139 @@ private boolean isUrl(String potentialUrl) {
+   * @param stageContext the stage context
    * @return
    */
   @VisibleForTesting
-  private Map<String, Object> createAugmentedRequest(Map<String, Object> trigger, Map<String, Object> buildInfo, Map<String, Object> request, boolean allowMissingPackageInstallation) {
-    Map<String, Object> artifactSourceBuildInfo = getArtifactSourceBuildInfo(trigger);
-    List<Map<String, Object>> triggerArtifacts = Optional.ofNullable((List<Map<String, Object>>) artifactSourceBuildInfo.get("artifacts")).orElse(emptyList());
-    List<Map<String, Object>> buildArtifacts = Optional.ofNullable((List<Map<String, Object>>) buildInfo.get("artifacts")).orElse(emptyList());
-
-    if (request.get("package") == null || request.get("package").equals("") || isUrl(request.get("package").toString())) {
-      return request;
+  private Map<String, Object> createAugmentedRequest(Map<String, Object> trigger,
+                                                     Map<String, Object> buildInfoCurrentExecution,
+                                                     Map<String, Object> stageContext,
+                                                     boolean allowMissingPackageInstallation) {
+    Map<String, Object> triggerBuildInfo = getBuildInfoFromTriggerOrParentTrigger(trigger);
+    List<Map<String, Object>> triggerArtifacts = Optional.ofNullable((List<Map<String, Object>>) triggerBuildInfo.get("artifacts")).orElse(emptyList());
+    List<Map<String, Object>> buildArtifacts = Optional.ofNullable((List<Map<String, Object>>) buildInfoCurrentExecution.get("artifacts")).orElse(emptyList());
+
+    if (stageContext.get("package") == null || stageContext.get("package").equals("") || isUrl(stageContext.get("package").toString())) {
+      return stageContext;
     }
 
-    if (buildInfo.isEmpty() || buildArtifacts.isEmpty()) {
+    if (buildInfoCurrentExecution.isEmpty() || buildArtifacts.isEmpty()) {
       Optional<Map<String, Object>> parentBuildInfo = Optional
         .ofNullable((Map) trigger.get("parentExecution"))
         .map(it -> (Map) it.get("trigger"))
         .map(it -> (Map<String, Object>) it.get("buildInfo"));
-      if (triggerArtifacts.isEmpty() && (trigger.get("buildInfo") != null || parentBuildInfo.isPresent())) {
+      if (triggerArtifacts.isEmpty() && (trigger.get("buildInfo") != null || parentBuildInfo.isPresent()) && artifacts.isEmpty()) {
         throw new IllegalStateException("Jenkins job detected but no artifacts found, please archive the packages in your job and try again.");
       }
     }
 
-    if (buildArtifacts.isEmpty() && triggerArtifacts.isEmpty()) {
-      return request;
+    if (buildArtifacts.isEmpty() && triggerArtifacts.isEmpty() && artifacts.isEmpty()) {
+      return stageContext;
     }
 
     List<String> missingPrefixes = new ArrayList<>();
     String fileExtension = format(".%s", packageType);
 
-    // There might not be a request.package so we look for the package name from either the buildInfo or trigger
```

Comment:
Above we say ```if (stageContext.get("package") == null || stageContext.get("package").equals("") || isUrl(stageContext.get("package").toString())) {
      return stageContext;
    }```

That indicates to me that we will have already failed if there are no artifacts specified. Please let me know if I'm missing something and this actually cannot be removed!